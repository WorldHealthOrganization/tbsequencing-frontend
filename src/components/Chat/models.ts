export interface IUserMessage {
  email: string,
  username: string,
  firstName: string,
  lastName: string
}
