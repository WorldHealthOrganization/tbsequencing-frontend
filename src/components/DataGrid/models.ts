export interface IColumn<T> {
  name: keyof T;
  header: string;
}

export enum GridType {
  Gene = 'gene',
  Drug = 'drug',
}

export enum ScrollType {
  Infinite = 'infinite',
  Regular = 'regular',
}

export type SortOrder = 'asc' | 'desc' | undefined;

export interface ISortState {
  colIndex?: number
  order?: SortOrder
  colID?: string
}

export interface IColumnFilter {
  id: string;
  value: number | string;
}

export interface BaseColumns {
  variantName: string;
  variantGrade: string;
  consequence: string;
  globalFrequency: string;
  totalCounts: number;
  resistantCount: number;
  susceptbleCount: number;
  intermediateCount: number;
  nucleodicAnnName: string;
  proteicAnnName: string;
  drug: number;
  drugName: string;
}

export interface DrugTableColumns extends BaseColumns {
  geneName: string;
}

export interface GeneTableColumns extends BaseColumns {
  drug: number;
}
