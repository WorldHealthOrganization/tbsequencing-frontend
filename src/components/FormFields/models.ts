import {
  ControllerRenderProps, FieldError, FieldName, FieldValues,
} from 'react-hook-form';
import { InputProps } from '@mui/material';
import { AbstractIField, CustomInputType } from '../../features/auth/components/registration/types';

export interface IBaseControlProps<IInput> extends
  AbstractIField<IInput>,
  Pick<ControllerRenderProps<FieldValues, FieldName<IInput>>, 'value' | 'onChange' | 'onBlur' > {
  isError: boolean,
  error: FieldError,
  fullWidth?: boolean,
  required?: boolean,
}

export interface ICheckboxProps<IInput> extends IBaseControlProps<IInput> {

}

export interface ITextAreaProps<IInput> extends IBaseControlProps<IInput> {

}

export interface ITextControlProps<IInput> extends IBaseControlProps<IInput> {
  type: CustomInputType
  InputProps?: Partial<InputProps>
}

export interface ISwitchControlProps<IInput> extends IBaseControlProps<IInput> {
  isActive?: boolean;
}

export interface ISelectControlProps<IInput> extends IBaseControlProps<IInput> {

}
