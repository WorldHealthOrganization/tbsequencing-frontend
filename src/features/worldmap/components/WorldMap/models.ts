export type MapType = 'macroRegions' | 'heatmap';
