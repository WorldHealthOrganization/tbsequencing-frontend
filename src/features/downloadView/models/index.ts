import { IDrug } from '../../../services/drugsApi/models';

export interface IExtendedDrug extends IDrug {
  isChecked: boolean;
}
