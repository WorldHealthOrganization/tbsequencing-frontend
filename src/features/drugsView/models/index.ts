export enum DrugStatus {
  Susceptible = 'susceptible',
  Resistant = 'resistant',
  Intermediate = 'intermediate',
  RatioSusceptible = 'ratioSusceptible',
  RatioResistant = 'ratioResistant',
  RatioIntermediate = 'ratioIntermediate',
}

export enum ResistantType {
  Phenotypic = 'Pheno',
  Genotypic = 'Geno',
}

export enum ChartDataType {
  Ratio = 'ratio',
  Count = 'count',
}

export enum Regions {
  ALL = 'ALL_REGIONS',
  AFRICA = 'AFR',
  EUROPE = 'EUR',
  AMERICA = 'AMR',
  EASTERN_MEDITERRANEAN = 'EMR',
  WESTERN_PACIFIC = 'WPR',
  SOUTH_EAST_ASIA = 'SEA',
}
