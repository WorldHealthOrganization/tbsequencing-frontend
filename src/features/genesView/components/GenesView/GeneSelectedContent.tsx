/** @jsxImportSource @emotion/react */
import React, { useEffect, useState } from 'react';
import GenomeContextWrapper from '../GenomeContextWrapper';
import H2 from '../../../../components/typography/H2';
import AssociationTable, { LeftColumnButton } from '../AssociationTable';
import { container, genesViewStyles, gridContainerStyles } from './styles';
import GnomadVariants from '../../../../components/GnomadVariants';
import { IGene, IVariant } from '../../../../services/genesApi/models';
import {
  useGetGenesByIdQuery,
  useGetGenesTableDataQuery,
  useGetGenomeContextQuery,
} from '../../../../services/genesApi/genesApi';
import { useTableData } from '../../../drugsView/hooks/useTableData';
import DataGrid from '../../../../components/DataGrid';
import {
  GeneTableColumns,
  GridType,
  IColumn,
  ScrollType,
} from '../../../../components/DataGrid/models';
import { IVariantLocal } from '../../models';
import { formatDrugsTableGenesView, formatGenesOverviewData, getGenomeContextLength } from './utils';
import GeneOverview from '../GeneOverview';
import { IDrug, TableData } from '../../../../services/drugsApi/models';
import { IAssociationTableData } from '../AssociationTable/AssociationTable';
import { useNavigateWithParams } from '../../../../hooks/useNavigateWithParams';
import { useGetGeneAssociationsByIDQuery } from '../../../../services/drugsApi/drugsApi';

interface IGeneSelectedContentProps {
  selectedGene: IGene
}

const columns: IColumn<GeneTableColumns>[] = [
  { name: 'variantName', header: 'Variant' },
  { name: 'nucleodicAnnName', header: 'Nucleic Annotation' },
  { name: 'proteicAnnName', header: 'Proteic Annotation' },
  { name: 'consequence', header: 'Consequence' },
  { name: 'variantGrade', header: 'Grade' },
  { name: 'globalFrequency', header: 'Global Frequency' },
  { name: 'totalCounts', header: 'Total Samples' },
  { name: 'resistantCount', header: 'Resistant Count' },
  { name: 'susceptbleCount', header: 'Susceptible Count' },
  { name: 'intermediateCount', header: 'Intermediate Count' },
];

const columnsWithoutStats: IColumn<GeneTableColumns>[] = [
  { name: 'variantName', header: 'Variant' },
  { name: 'nucleodicAnnName', header: 'Nucleic Annotation' },
  { name: 'proteicAnnName', header: 'Proteic Annotation' },
  { name: 'consequence', header: 'Consequence' },
  { name: 'variantGrade', header: 'Grade' },
  { name: 'globalFrequency', header: 'Global Frequency' },
  { name: 'totalCounts', header: 'Total Samples' },
];

const GeneSelectedContent = ({ selectedGene }: IGeneSelectedContentProps) => {
  const navigate = useNavigateWithParams();
  const onClick = (item: IAssociationTableData) => {
    const params = new URLSearchParams({
      drugId: String(item.id),
      code: String(item.leftColValue),
      drugName: String(item.rightColValue),
    });
    navigate('/drugs', params);
  };

  const [selectedVariant, setSelectedVariant] = useState<IVariantLocal | {}>({});
  const [activeIndex, setActiveIndex] = useState<number>(-1);
  const [gridViewPortData, setGridViewPortData] = useState<number[]>([]);
  const [selectedResistantDrug, setSelectedResistantDrug] = useState<IDrug>();

  const { data: genesDetails, isLoading: isGeneDetailsLoading } = useGetGenesByIdQuery({
    geneDbCrossrefId: selectedGene.geneDbCrossrefId,
  });

  const { startPos, endPos } = getGenomeContextLength(genesDetails?.genesOverview);

  const { data: genomeContextData, isLoading: isGenomeContextLoading } = useGetGenomeContextQuery({
    startPos,
    endPos,
  }, { skip: !genesDetails?.genesOverview?.startPos || !genesDetails?.genesOverview?.endPos });

  const {
    data: associationsData,
    isLoading: isAssociationsLoading,
    isSuccess,
  } = useGetGeneAssociationsByIDQuery({
    geneDbCrossrefId: selectedGene.geneDbCrossrefId,
  });

  const {
    tableData = [],
    isTableDataLoading,
    filters,
    updateFilters,
  } = useTableData({
    useGetDataQuery: useGetGenesTableDataQuery,
    geneDbCrossrefId: selectedGene.geneDbCrossrefId,
    drugID: selectedResistantDrug?.drugId || associationsData?.[0]?.drugId,
    skipQuery: !isSuccess,
  });

  const drugsTableData = formatDrugsTableGenesView(associationsData);
  const genesOverviewFormattedData = formatGenesOverviewData(genesDetails?.genesOverview);

  useEffect(() => {
    const newActiveIndex = tableData.findIndex((data: TableData) => (
      data.variantName === (selectedVariant as IVariantLocal).variantName
    ));
    setActiveIndex(newActiveIndex);
  }, [tableData, selectedVariant]);

  const onVariantClick = (variant: any) => {
    setSelectedVariant(variant);
  };

  const onScroll = (visibleRowsRange: number[]) => {
    setGridViewPortData(visibleRowsRange);
  };

  const onChangeResistantDrug = (value: number) => {
    if (!associationsData?.length) return;
    const drug = associationsData.find((item) => item.drugId === value);
    setSelectedResistantDrug(drug);
  };

  return (
    <div css={container}>
      <GenomeContextWrapper
        selectedGene={genesDetails?.genesOverview}
        isLoading={isGenomeContextLoading}
        data={genomeContextData}
        maxDomain={endPos}
        minDomain={startPos}
      />
      <H2 style={genesViewStyles.overviewTitle}>Overview</H2>
      <div css={genesViewStyles.tablesWrapper}>
        <GeneOverview data={genesOverviewFormattedData} isLoading={isGeneDetailsLoading} />
        <AssociationTable
          tableHeader="Gene-drug resistance associations"
          data={drugsTableData}
          isLoading={isAssociationsLoading}
          leftColumnItemComponent={LeftColumnButton}
          onClick={onClick}
          noDataText="No associated data with this gene"
        />
      </div>
      <div css={genesViewStyles.gnomadWrapper}>
        <GnomadVariants
          selectedGene={selectedGene}
          variants={tableData as unknown as IVariant[]}
          isLoading={isTableDataLoading}
          variantsInTable={gridViewPortData}
          onClick={onVariantClick}
        />
      </div>
      <div css={genesViewStyles.gridWrapper}>
        <DataGrid<GeneTableColumns>
          containerStyles={gridContainerStyles}
          columns={associationsData?.length ? columns : columnsWithoutStats}
          isLoading={isTableDataLoading}
          filterValues={filters}
          onFilterChange={updateFilters}
          dataSource={tableData}
          selectedDropdownItem={selectedGene}
          selectedRow={activeIndex}
          type={GridType.Gene}
          scrollType={ScrollType.Infinite}
          resistantDrugs={associationsData}
          selectedResistantDrug={selectedResistantDrug || associationsData?.[0]}
          onChangeResistantDrug={onChangeResistantDrug}
          onScroll={onScroll}
        />
      </div>
    </div>
  );
};

export default GeneSelectedContent;
