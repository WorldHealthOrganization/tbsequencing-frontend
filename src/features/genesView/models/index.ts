import { skipToken } from '@reduxjs/toolkit/query';
import React from 'react';
import { IGene, IVariant } from '../../../services/genesApi/models';

export type AutocompleteParams = typeof skipToken | { search?: string };

export interface IGenomeContextWrapperProps {
  data?: IGene[];
  isLoading?: boolean;
  selectedGene?: IGenesOverview;
  minDomain?: number;
  maxDomain?: number;
}

export interface IGenomeContextProps extends IGenomeContextWrapperProps {
  width: number;
}

export interface IVariantLocal extends IVariant {
  pos: number;
}

export interface ISearchResult {
  geneName: string;
  geneDbCrossrefId: number;
  locusTag: string;
}

export interface ISearchCount extends ISearchResult {
  counter: number;
}

export interface ISearchRecent extends ISearchResult {
  counter: string;
  date: Date | string;
}

export interface IGenesOverview extends IGene {
  geneDescription: string;
  geneType: string;
  ncbiId: string;
  proteinLength: number;
}

export interface IGenesOverviewTableData {
  title: string;
  value: string | number | null;
  element?: React.ReactNode;
}

export interface IGenesOverviewResponse {
  genes: IGene;
  genesOverview: IGenesOverview
}
