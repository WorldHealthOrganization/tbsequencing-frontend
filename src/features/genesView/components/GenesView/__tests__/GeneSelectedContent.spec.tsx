import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { getTreeJSONWithRedux, render } from '../../../../../utils/testUtils/jestUtils';
import GeneSelectedContent from '../GeneSelectedContent';
import { IGene } from '../../../../../services/genesApi/models';

const mockGene: IGene = {
  endPos: 0,
  startPos: 0,
  geneDbCrossrefId: 1,
  geneName: 'rpoC',
  locusTag: 'RVca92',
  id: 1,
  strand: 1,
};

describe('GeneSelectedContent', () => {
  test('renders GeneSelectedContent component without error', () => {
    render(
      <BrowserRouter>
        <GeneSelectedContent selectedGene={mockGene} />
      </BrowserRouter>,
    );
  });

  test('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <BrowserRouter>
        <GeneSelectedContent selectedGene={mockGene} />
      </BrowserRouter>,
    );

    expect(tree).toMatchSnapshot();
  });
});
