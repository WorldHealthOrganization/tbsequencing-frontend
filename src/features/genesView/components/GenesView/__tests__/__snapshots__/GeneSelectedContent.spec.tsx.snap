// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneSelectedContent Should render correctly 1`] = `
<div
  className="css-xy18v1-GeneSelectedContent"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1ag5kwf-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
  <p
    className="MuiTypography-root MuiTypography-body1 css-rari9s-MuiTypography-root-BaseText"
  >
    Overview
  </p>
  <div
    className="css-3g6t7a-GeneSelectedContent"
  >
    <div
      className="css-9j8ib4-AppPaper"
    >
      <div />
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1ag5kwf-MuiCircularProgress-root"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
    <div
      className="css-9j8ib4-AppPaper"
    >
      <div>
        <p
          className="MuiTypography-root MuiTypography-body1 css-5th3yp-MuiTypography-root-BaseText"
        >
          Gene-drug resistance associations
        </p>
      </div>
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1ag5kwf-MuiCircularProgress-root"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="css-2q0ddi-GeneSelectedContent"
  >
    <div
      className="css-gsnibn-AppPaper"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-nwtw2h-MuiTypography-root-BaseText"
      >
        Variants
      </p>
      No results matching your request
      <div
        className="css-x4blsr-GnomadVariants"
      >
        <div
          className="css-8t833m-GnomadVariants"
        >
          <div
            className="css-mbzhaz-GnomadVariants"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
          >
             UPSTREAM 
          </p>
        </div>
        <div
          className="css-1v536bo-GnomadVariants"
        >
          <div
            className="css-1dhfc98-GnomadVariants"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
          >
             MISSENSE 
          </p>
        </div>
        <div
          className="css-dcndn1-GnomadVariants"
        >
          <div
            className="css-1q4a6hc-GnomadVariants"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
          >
             SYNONYMOUS 
          </p>
        </div>
        <div
          className="css-f1cibi-GnomadVariants"
        >
          <div
            className="css-wgv6en-GnomadVariants"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
          >
             INSERTION/DELETION 
          </p>
        </div>
        <div
          className="css-1j3np5j-GnomadVariants"
        >
          <div
            className="css-zuj9il-GnomadVariants"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
          >
             LOSS-OF-FUNCTION 
          </p>
        </div>
        <div
          className="css-euhmwu-GnomadVariants"
        >
          <div
            className="css-ud7xso-GnomadVariants"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
          >
             OTHER 
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1ls4gnc-GeneSelectedContent"
  >
    <div>
      <div
        className="css-1dqm8h-AppPaper"
      >
        <div
          className="css-zuqjgc-ExportButton"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-5th3yp-MuiTypography-root-BaseText"
          >
            Export CSV Data
          </p>
          <div
            className="css-uhrw2b-ExportButton"
          >
            <button
              className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-sfprtt-MuiButtonBase-root-MuiButton-root-AppButton"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <span
                  aria-hidden={true}
                  className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
                >
                  file_download
                </span>
              </span>
              <p
                className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
              >
                Download
              </p>
            </button>
            <a
              download="rpoC"
              href="data:text/csv;charset=utf-8,ï»¿\\"Variant\\",\\"Nucleic Annotation\\",\\"Proteic Annotation\\",\\"Consequence\\",\\"Grade\\",\\"Global Frequency\\",\\"Total Samples\\""
              onClick={[Function]}
              target="_self"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <div
          className="css-1ou1qes-DataGrid"
        >
          <table
            className="MuiTable-root css-adcf9a-MuiTable-root"
            role={null}
          >
            <thead
              className="MuiTableHead-root css-cwxlwz-MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1fhv7sw-MuiTableCell-root"
                  scope="col"
                >
                  <div
                    className="css-1qyj6ee-Thead"
                  >
                    <span
                      className="css-qcmszo-TableHeader"
                    >
                      Variant
                    </span>
                    <div
                      className="css-trc57z-Thead"
                    >
                      <button
                        className="css-13hx71u-Thead"
                        data-testid="filter-button-variantName"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1o8iwtf-MuiSvgIcon-root"
                          color="#009ADE"
                          data-testid="SearchIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1fhv7sw-MuiTableCell-root"
                  scope="col"
                >
                  <div
                    className="css-1qyj6ee-Thead"
                  >
                    <span
                      className="css-qcmszo-TableHeader"
                    >
                      Nucleic Annotation
                    </span>
                    <div
                      className="css-trc57z-Thead"
                    >
                      <button
                        className="css-13hx71u-Thead"
                        data-testid="filter-button-nucleodicAnnName"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1o8iwtf-MuiSvgIcon-root"
                          color="#009ADE"
                          data-testid="SearchIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1fhv7sw-MuiTableCell-root"
                  scope="col"
                >
                  <div
                    className="css-1qyj6ee-Thead"
                  >
                    <span
                      className="css-qcmszo-TableHeader"
                    >
                      Proteic Annotation
                    </span>
                    <div
                      className="css-trc57z-Thead"
                    >
                      <button
                        className="css-13hx71u-Thead"
                        data-testid="filter-button-proteicAnnName"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1o8iwtf-MuiSvgIcon-root"
                          color="#009ADE"
                          data-testid="SearchIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1fhv7sw-MuiTableCell-root"
                  scope="col"
                >
                  <div
                    className="css-1qyj6ee-Thead"
                  >
                    <span
                      className="css-qcmszo-TableHeader"
                    >
                      Consequence
                    </span>
                    <div
                      className="css-trc57z-Thead"
                    >
                      <button
                        className="css-13hx71u-Thead"
                        data-testid="filter-button-consequence"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1o8iwtf-MuiSvgIcon-root"
                          color="#009ADE"
                          data-testid="SearchIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1fhv7sw-MuiTableCell-root"
                  scope="col"
                >
                  <div
                    className="css-1qyj6ee-Thead"
                  >
                    <span
                      className="css-qcmszo-TableHeader"
                    >
                      Grade
                    </span>
                    <div
                      className="css-trc57z-Thead"
                    >
                      <button
                        className="css-13hx71u-Thead"
                        data-testid="filter-button-variantGrade"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1o8iwtf-MuiSvgIcon-root"
                          color="#009ADE"
                          data-testid="SearchIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1fhv7sw-MuiTableCell-root"
                  scope="col"
                >
                  <div
                    className="css-1qyj6ee-Thead"
                  >
                    <span
                      className="css-qcmszo-TableHeader"
                    >
                      Global Frequency
                    </span>
                    <div
                      className="css-trc57z-Thead"
                    />
                  </div>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1fhv7sw-MuiTableCell-root"
                  scope="col"
                >
                  <div
                    className="css-1qyj6ee-Thead"
                  >
                    <span
                      className="css-qcmszo-TableHeader"
                    >
                      Total Samples
                    </span>
                    <div
                      className="css-trc57z-Thead"
                    />
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                role={null}
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-4b1vly-MuiTableCell-root"
                  colSpan={100}
                  rowSpan={100}
                >
                  No results matching your request
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
